File: Colas con Listas/Cola lineal.py
# queue_list.py
from collections import deque

def insertar(q):
    elemento = input("Ingrese el elemento: ")
    q.append(elemento)
    print("Elemento insertado correctamente.")

def eliminar(q):
    if not q:
        print("SUBDESBORDAMIENTO (UNDERFLOW)")
        return
    elemento = q.popleft()
    print("Elemento eliminado:", elemento)

def mostrar(q):
    if not q:
        print("La cola está vacía.")
        return
    print("Elementos en la cola:")
    for elem in q:
        print(elem)

def main():
    q = deque()
    opcion = None
    while opcion != "4":
        print("\n*************** MENU PRINCIPAL ***************")
        print("1. Insertar un elemento")
        print("2. Eliminar un elemento")
        print("3. Mostrar la cola")
        print("4. Salir")
        opcion = input("Ingrese su opción: ").strip()

        if opcion == "1":
            insertar(q)
        elif opcion == "2":
            eliminar(q)
        elif opcion == "3":
            mostrar(q)
        elif opcion == "4":
            print("Saliendo del programa...")
        else:
            print("Opción inválida. Intente nuevamente.")

if __name__ == "__main__":
    main()

File: Colas con Listas/Cola lineal.cs
// QueueList.cs
using System;
using System.Collections.Generic;

class QueueList {
    static void Insertar(Queue<int> q) {
        Console.Write("Ingrese el elemento (entero): ");
        string input = Console.ReadLine();
        int elemento;
        while (!int.TryParse(input, out elemento)) {
            Console.Write("Por favor ingrese un número entero: ");
            input = Console.ReadLine();
        }
        q.Enqueue(elemento);
        Console.WriteLine("Elemento insertado correctamente.");
    }

    static void Eliminar(Queue<int> q) {
        if (q.Count == 0) {
            Console.WriteLine("SUBDESBORDAMIENTO (UNDERFLOW)");
            return;
        }
        int elemento = q.Dequeue();
        Console.WriteLine("Elemento eliminado: " + elemento);
    }

    static void Mostrar(Queue<int> q) {
        if (q.Count == 0) {
            Console.WriteLine("La cola está vacía.");
            return;
        }
        Console.WriteLine("Elementos en la cola:");
        foreach (var elem in q) {
            Console.WriteLine(elem);
        }
    }

    static void Main() {
        var q = new Queue<int>();
        string opcion = "";
        while (opcion != "4") {
            Console.WriteLine("\n*************** MENU PRINCIPAL ***************");
            Console.WriteLine("1. Insertar un elemento");
            Console.WriteLine("2. Eliminar un elemento");
            Console.WriteLine("3. Mostrar la cola");
            Console.WriteLine("4. Salir");
            Console.Write("Ingrese su opción: ");
            opcion = Console.ReadLine().Trim();

            switch (opcion) {
                case "1":
                    Insertar(q);
                    break;
                case "2":
                    Eliminar(q);
                    break;
                case "3":
                    Mostrar(q);
                    break;
                case "4":
                    Console.WriteLine("Saliendo del programa...");
                    break;
                default:
                    Console.WriteLine("Opción inválida. Intente nuevamente.");
                    break;
            }
        }
    }
}
